Mesh_2/include/CGAL/IO/vtk_io_2.h

```diff
@@ -0,0 +1,312 @@
+      connectivity_table.push_back(V[fit->vertex(1)]);
+    }
+  }
+  write_vector<std::size_t>(os,connectivity_table);
+  write_vector<std::size_t>(os,offsets);
+  write_vector<unsigned char>(os,cell_type);
+}
+
+// writes the points tags before binary data is appended
+template <class Tr>
+void 
+write_points_tag(std::ostream& os,
+                 const Tr & tr,
+                 std::map<typename Tr::Vertex_handle, std::size_t> & V,
+                 bool binary,
+                 std::size_t& offset) 
+{
+  std::size_t dim = 2;
+  typedef typename Tr::Finite_vertices_iterator Finite_vertices_iterator;
+  typedef typename Tr::Geom_traits Gt;
+  typedef typename Gt::FT FT;
+
+  std::size_t inum = 0;
+  std::string format = binary ? "appended" : "ascii";
+  std::string type = (sizeof(FT) == 8) ? "Float64" : "Float32";
+
+  os << "    <Points>\n"
+     << "      <DataArray type =\"" << type << "\" NumberOfComponents=\"3\" format=\""
+     << format;
+
+  if (binary) {
+    os << "\" offset=\"" << offset << "\"/>\n";    
+    offset += 3 * tr.number_of_vertices() * sizeof(FT) + sizeof(std::size_t);
+    // dim coords per points + length of the encoded data (size_t)
+  }
+  else {
+    os << "\">\n";  
+    for( Finite_vertices_iterator vit = tr.finite_vertices_begin();
+         vit != tr.finite_vertices_end();
+         ++vit)
+    {
+      V[vit] = inum++;
+        os << vit->point()[0] << " ";
+        os << vit->point()[1] << " ";
+        if(dim == 3) 
+          os << vit->point()[2] << " ";
+        else
+          os << 0.0 << " ";
+      }
+    os << "      </DataArray>\n";
+  }
+  os << "    </Points>\n";
+}
+
+// writes the points appended data at the end of the .vtu file 
+template <class Tr>
+void
+write_points(std::ostream& os,
+	     const Tr & tr,
+	     std::map<typename Tr::Vertex_handle, 
+             std::size_t> & V)
+{
+  std::size_t dim = 2;
+  typedef typename Tr::Finite_vertices_iterator Finite_vertices_iterator;
+  typedef typename Tr::Geom_traits Gt;
+  typedef typename Gt::FT FT;
+
+  std::size_t inum = 0;
+  std::vector<FT> coordinates;
+  for( Finite_vertices_iterator vit = tr.finite_vertices_begin();
+       vit != tr.finite_vertices_end();
+       ++vit)
+    {
+      V[vit] = inum++;  // binary output => the map has not been filled yet
+      coordinates.push_back(vit->point()[0]);
+      coordinates.push_back(vit->point()[1]);
+      coordinates.push_back(dim == 3 ? vit->point()[2] : 0.0);
+    }
+  write_vector<FT>(os,coordinates);
+}
+
+//!
+//! \brief write_unstructured_grid_2 writes the content of a `CDT` in the .vtu
+//! XML format.
+//! 
+//! \tparam CDT a `Constrained_Delaunay_triangulation_2`.
+//! 
+//! \param os a `std::ostream`.
+//! \param cdt an instance of `CDT` to be written.
+//! \param binary decides if the data should be written in binary(`true`)
+//!   or in ASCII(`false`).
+//!
+template <class CDT>
+void write_unstructured_grid_2(std::ostream& os,
```

Comment:
I think the export of a `CDT_2` to a VTU file should also output the constrained edges, if possible. It seems the [`vtkUnstructuredGrid`] can mix type of "cells".

[`vtkUnstructuredGrid`]: https://www.vtk.org/doc/nightly/html/classvtkUnstructuredGrid.html